agent.go

```diff
@@ -35,16 +35,16 @@ func AgentRequested(sess Session) bool {
 
 // NewAgentListener sets up a temporary Unix socket that can be communicated
 // to the session environment and used for forwarding connections.
-func NewAgentListener() (net.Listener, error) {
-	dir, err := ioutil.TempDir("", agentTempDir)
+func NewAgentListener() (net.Listener, string, error) {
```

Comment:
Looks like this'll break one of the examples, could you update that as well please? 
https://github.com/loft-sh/ssh/blob/14711385e6a2bdca49a2446e4d134a8b4e46011b/_examples/ssh-forwardagent/forwardagent.go#L15